display.go

```diff
@@ -12,16 +12,79 @@ import (
 	"fyne.io/fyne/v2/canvas"
 	"fyne.io/fyne/v2/container"
 	"fyne.io/fyne/v2/driver/desktop"
+	"fyne.io/fyne/v2/widget"
+	"tinygo.org/x/drivers/touch"
 )
 
+type touchWidget struct {
+	widget.BaseWidget
+	obj fyne.CanvasObject
+	d   *Device
+}
+
+func (s *touchWidget) CreateRenderer() fyne.WidgetRenderer {
+	return widget.NewSimpleRenderer(s.obj)
+}
+
+func (s *touchWidget) setTouchPoint(p fyne.Position) {
+
+	x := int(p.X)
+	y := int(p.Y)
+
+	//Clamp X & Y values to width and height. If the pointer is moved outside the window these values may be negative or larger than the window
+	if x < 0 {
+		x = 0
+	} else if x > s.d.Width {
+		x = s.d.Width
+	}
+
+	if y < 0 {
+		y = 0
+	} else if y > s.d.Height {
+		y = s.d.Height
+	}
```

Comment:
I think it should be from `0` to `s.d.Width-1`.
If x is s.d.Width, then s.d.TouchPoint.X will be 0x10000 instead of 0xFFFF.

* width : 0 to s.d.Width - 1
* height : 0 to s.d.Height - 1

Reply:
I've made this change. I also set Z to 0xFFFF to keep it inside the 16bit range.